python/cudf/cudf/_libxx/copying.pyx

```diff
@@ -127,11 +127,15 @@ def copy_range(Column input_column,
                            input_begin, input_end, target_begin)
 
 
-def gather(Table source_table, Column gather_map):
+def gather(Table source_table, Column gather_map, index=True):
```

Comment:
Can we make this `keep_index` similar to: https://github.com/rapidsai/cudf/blob/branch-0.14/python/cudf/cudf/_libxx/copying.pyx#L247

and propagate it upwards as `keep_index` as well?